analyzers/tests/SonarAnalyzer.TestFramework/Verification/DiagnosticVerifier.cs

```diff
@@ -99,40 +96,23 @@ public static class DiagnosticVerifier
             onlyDiagnostics ??= Array.Empty<string>();
             var supportedDiagnostics = analyzer
                 .SelectMany(x => x.SupportedDiagnostics.Select(d => d.Id))
-                .Concat(new[] { AD0001 })
-                .Select(x => new KeyValuePair<string, ReportDiagnostic>(x, Severity(x)))
-                .ToArray();
-
-            var ids = supportedDiagnostics.Select(x => x.Key).ToHashSet();
-
+                .ToImmutableDictionary(x => x, Severity)
+                .Add(AD0001, ReportDiagnostic.Error);
```

Comment:
I cannot find a better way to do this but it's a bit hard to the eye making an immutable dictionary and then re-creating it to add one entry 😅 .

Reply:
This is the way immutable directories are build. They always get created then then entries are added as a AVL tree (yes, immutable dictionary is not a dictionary, but a tree)

Another example:
https://github.com/SonarSource/sonar-dotnet/blob/48f9399972fefba70fca547dcfead9a1e0ecb0ae/analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs#L41